core/trino-main/src/main/java/io/trino/server/TaskResource.java

```diff
@@ -138,11 +142,19 @@ public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo)
     @Produces(MediaType.APPLICATION_JSON)
     public void createOrUpdateTask(
             @PathParam("taskId") TaskId taskId,
+            @QueryParam("instanceId") String expectedInstanceId,
             TaskUpdateRequest taskUpdateRequest,
             @Context UriInfo uriInfo,
             @Suspended AsyncResponse asyncResponse)
     {
         requireNonNull(taskUpdateRequest, "taskUpdateRequest is null");
+        if (!nodeInstanceId.equals(expectedInstanceId)) {
+            asyncResponse.resume(Response.status(Status.BAD_REQUEST)
+                    .type(MediaType.TEXT_PLAIN_TYPE)
+                    .entity("Node instance ID in the request [%s] does not match this node instance ID [%s]. This may indicate a late request targeted for a previously running node with same IP address.".formatted(expectedInstanceId, nodeInstanceId))
+                    .build());
+            return;
```

Comment:
nit: I would slighttly prefer to extract this repeated block as method which returns boolean and just have sth
like
```
if (!checkInstanceIdValid(asyncResonse, expectedInstanceId) {
   return;
}
```

Reply:
will do!